src/Microsoft.ML.ImageAnalytics/Imager.cs

```diff
@@ -0,0 +1,420 @@
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// Provide interfaces for imaging operations.
+    /// </summary>
+    public class Imager : IDisposable
+    {
+        private SKBitmap _image;
+
+        /// <summary>
+        /// Create a new imager instance from a stream.
+        /// </summary>
+        /// <param name="imageStream">The stream to create the image from.</param>
+        public Imager(Stream imageStream)
+        {
+            if (imageStream is null)
+            {
+                throw new ArgumentNullException(nameof(imageStream));
+            }
+
+            _image = SKBitmap.Decode(imageStream);
+            if (_image is null)
+            {
+                throw new ArgumentException($"Invalid input stream contents", nameof(imageStream));
+            }
+
+            EnsureSupportedPixelFormat();
+        }
+
+        /// <summary>
+        /// Create a new imager instance from image file.
+        /// </summary>
+        /// <param name="imagePath">The image file path to create the image from.</param>
+        public Imager(string imagePath)
+        {
+            if (imagePath is null)
+            {
+                throw new ArgumentNullException(nameof(imagePath));
+            }
+
+            _image = SKBitmap.Decode(imagePath);
+            if (_image is null)
+            {
+                throw new ArgumentException($"Invalid path", nameof(imagePath));
+            }
+
+            EnsureSupportedPixelFormat();
+        }
+
+        private Imager(SKBitmap image)
+        {
+            _image = image;
+            EnsureSupportedPixelFormat();
+        }
+
+        private void EnsureSupportedPixelFormat()
+        {
+            Debug.Assert(_image is not null);
+
+            // Most of the time SkiaSharp create images with Bgra8888 or Rgba8888 pixel format.
+            if (_image.Info.ColorType != SKColorType.Bgra8888 && _image.Info.ColorType != SKColorType.Rgba8888)
+            {
+                if (!_image.CanCopyTo(SKColorType.Bgra8888))
+                {
+                    throw new InvalidOperationException("Unsupported image format.");
+                }
+
+                SKBitmap image1 = _image.Copy(SKColorType.Bgra8888);
+                _image.Dispose();
+                _image = image1;
+            }
+        }
+
+        /// <summary>
+        /// Create BRGA32 pixel format imager object from the pixel data buffer span.
+        /// </summary>
+        /// <param name="width">The width of the image in pixels.</param>
+        /// <param name="height">The height of the image in pixels.</param>
+        /// <param name="imagePixelData">The pixels data to create the image from.</param>
+        /// <returns>Imager object.</returns>
+        public static unsafe Imager CreateFromBgra32PixelData(int width, int height, Span<byte> imagePixelData)
```

Comment:
Should this be `ReadOnlySpan<byte>`?

Reply:
That is a good point. It should be ReadOnlySpan :-) thanks for the catch.

Reply:
done